unikraft/app: Pin ordering for libC's

craciunoiuc opened the pull request:
Because libraries are kept in a map, their ordering can't be guaranteed This comes in conflict with libC-type libraries, which need to sit first. This is not a problem related with them, and comes from other libraries using the `#include_next` syntax on them.

To fix this we pin either `musl` or `newlib` plus `pthread-embedded` to come first, if they exist.

<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

nderjung commented on unikraft/app/application.go:
```diff
@@ -276,7 +276,33 @@ func (app application) IsConfigured(tc target.Target) bool {
 	var libraries []string
 
-	for _, library := range app.libraries {
+	//  _______ ____  _____   ____
+	// |__   __/ __ \|  __ \ / __ \
+	//    | | | |  | | |  | | |  | |
+	//    | | | |  | | |  | | |  | |
+	//    | | | |__| | |__| | |__| |
+	//    |_|  \____/|_____/ \____/
```
Lol, you can remove this ;)

nderjung commented on unikraft/app/application.go:
```diff
@@ -276,7 +276,27 @@ func (app application) IsConfigured(tc target.Target) bool {
 	var libraries []string
 
-	for _, library := range app.libraries {
+	// TODO dirty hack to get build to work regardless of order of libraries
+	// Remove this once fixed in the main repository
```
Can you update the text:
```
// TODO: This is a temporary solution to fix an ordering issue with regard to
// syscall availability from a libc (which should be included first).  Long-term
// solution is to determine the library order by generating a DAG via KConfig
// parsing.
```

craciunoiuc committed:
unikraft/app: Pin ordering for libC's

Because libraries are kept in a map, their ordering can't be guaranteed
This comes in conflict with libC-type libraries, which need to sit
first. This is not a problem related with them, and comes from other
libraries using the `#include_next` syntax on them.

To fix this we pin either `musl` or `newlib` plus `pthread-embedded` to
come first, if they exist.

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

nderjung reviewed (approved):
Looks good to me, thanks!

Approved-by: Alexander Jung <alex@unikraft.io>
